shell/platform/android/io/flutter/embedding/android/FlutterTextureView.java

```diff
@@ -83,7 +83,13 @@ public boolean onSurfaceTextureDestroyed(@NonNull SurfaceTexture surface) {
           if (isAttachedToFlutterRenderer) {
             disconnectSurfaceFromRenderer();
           }
-
+          
+          // Definitively release the surface to avoid leaked closeables, just in case
+          if (renderSurface != null) {
+            renderSurface.release();
```

Comment:
this change will need a test. @bparrishMines was going to add it.

we could create a new activity with the right strict mode flags, and trigger this path. For reference, there's an existing activity in https://github.com/flutter/engine/blob/master/testing/scenario_app/android/app/src/main/AndroidManifest.xml#L40.

Reply:
How does this path get triggered? Is it through a `TextureRegistry`?

Reply:
To trigger this path, an extra intent parameter `background_mode`  must be set to `opaque`.

Reply:
To trigger this path, an extra intent parameter `background_mode`  must be set to `opaque` when the activity is launched. Then, toggle the app switcher to move the app to the background/foreground.

If this repros the crash, then the test could add a new activity that sets the right vm policy set `onCreate`, then starts this activity with the intent (as described above), then stop the activity.